core/src/hir/attrs.rs

```diff
@@ -158,11 +155,15 @@ fn maybe_error_unsupported(
     }
 }
 impl Attrs {
+    /// `location_validator` represents a callback passed in by `lowering.rs` methods to check if any attributes of a given location are allowed.
+    ///
+    /// Right now this is just used for [`Attrs::allowed_on_param_or_field`], but could be used for other attributes in specific locations.
     pub fn from_ast(
         ast: &ast::Attrs,
         validator: &(impl AttributeValidator + ?Sized),
         parent_attrs: &Attrs,
         errors: &mut ErrorStore,
+        location_validator: Option<fn(syn::Ident) -> Result<(), LoweringError>>,
```

Comment:
issue: let's not do it this way, this feels overly complicated


instead can we add Param and Field to AttributeContext, and perform this validation during attribute validation?

Reply:
there's also a good argument to be made for `from_ast` to have its own simpler AttributeContext enum, but it's probably easier to perform further validation in the preexisting validation pass

Reply:
Sure, will adjust now